CleverPush/Source/CleverPushInstance.m

```diff
@@ -1116,7 +1125,7 @@ - (void)registerDeviceToken:(id)newDeviceToken onSuccess:(CPResultSuccessBlock)s
 
         if (@available(iOS 10.0, *)) {
             [[UNUserNotificationCenter currentNotificationCenter] getNotificationSettingsWithCompletionHandler:^(UNNotificationSettings* settings) {
-                if (settings.authorizationStatus == UNAuthorizationStatusAuthorized) {
+                if (settings.authorizationStatus == UNAuthorizationStatusAuthorized || (!autoRequestNotificationPermission && settings.authorizationStatus == UNAuthorizationStatusNotDetermined)) {
```

Comment:
Why is this condition required?

Reply:
After getting the device token,

Earlier, it was like this:

`if (settings.authorizationStatus == UNAuthorizationStatusAuthorized) { }`
 if the notification permission had been granted, then only the subscribe API was being called.

But if the user had not given the notification permission and the `autoRequestNotificationPermission` flag was set to false, then we need to call the `subscribe()` method. That's why I keep the condition that if the user sets the `autoRequestNotificationPermission` to false and notification permission has not been asked for, then we need to call the `subscribe()` method.

Reply:
No, in this case we do not want to call the subscribe method. The user can use setIgnoreDisabledNotificationPermission to enable this behaviour.
setAutoRequestNotificationPermission(false) should just NOT auto-request the permission - no other logic changes.

Reply:
Please also go through the rest of the code and make sure no other logic changes when calling / not calling subscribe are involved in this PR.